docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,117 @@
+# Using CNI Interfaces and libvirt for networking
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+Today networking in KubeVirt is not integrated with Kubernetes, a user can
+rather choose to create a specific VM Spec definition to create a particular
+libvirt domxml, to reuse an existing physical device.
+But this is obviously suboptimal
+
+- It is not integrated with Kubernetes
+- It does not provide connectivity with pods
+- It requires to specify host specific bits (the network device to use)
+
+The purpose of this document is to describe one approach of how we can provide
+a minimal integrated solution, with few workarounds.
+
+**Note:** This approach is rough, and not optimized, but the rule of thumb was
+to find a robust solution which works with Kubernetes today.
+
+
+## Use-case
+
+The primary use-case is to allow VM to get connected to the pod network.
+
+
+## API
+
+The API is pretty clear: As usual an interface will be defined.
+To indicate that Pod interfaces shall be used for networking, a new type
+`pod-network` is introduced.
+All other fields, concerning the properties of the virtual NIC, can be used
+as usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+```
+
+
+## Implementation
+
+Prerequisites
+- libvirt is using Pod networking
+- virt-handler and libvirt are running in the same Pod or rather network
```

Comment:
This is the core reason why I suggested not directly running CNI from virt-handler, but instead putting it in its own daemon, which virt-handler talks to over REST or something similar. That way we don't have a dependency on what namespace virt-handler is in. With it tightly coupled we're creating problems for ourselves because we're inevitably going to hit case where some virt-handler features needs us to be in the non-host network namespace, while CNI requires host network.